packages/graphql-compiler/core/GraphQLParser.js

```diff
@@ -546,19 +546,15 @@ class GraphQLParser {
     if (argumentDirectives.length) {
       args = (argumentDirectives[0].arguments || []).map(arg => {
         const argValue = arg.value;
-        invariant(
-          argValue.kind === 'Variable',
-          'GraphQLParser: All @arguments() args must be variables, got %s. ' +
-            'Source: %s.',
-          argValue.kind,
-          this._getErrorContext(),
-        );
 
         return {
           kind: 'Argument',
           metadata: null,
           name: getName(arg),
-          value: this._transformVariable(argValue),
+          value:
+            argValue.kind === 'Variable'
+              ? this._transformVariable(argValue)
+              : this._transformValue(arg.value, null),
```

Comment:
Sorry if this is a noob question.
Will this change work well with `FlattenTransform` and `SkipRedundantNodeTransform` in case the literal values are resolved to be the the same?

Reply:
I added another test where the values are the same and they do get merged.